Clinical-Genomics/scout
# Additional check for user logged in status and requested resource before serving files using remote static

Fix #3302 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test - locally**:
1. On main branch, serve demo app (`scout --demo serve`)
2. Try to get a resource as a non-logged user via curl: `curl http://localhost:5000/remote/static?file=config.py`
3. The response should return the resource
4. Get the resource in the browser as a logged in user by using this URL: `http://localhost:5000/remote/static?file=config.py`
5. The response should return the resource

6. Switch to this branch and repeat 2 and 4
7. You should get a 403 code (forbidden) in both cases

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
